### PR TITLE
feat(exchange,binance, gate, bybit, okx, bitmex): add watchLiquidations and watchMyLiquidations, add tests for fetch and watchLiquidations

### PR DESCRIPTION
**Pull request by pcriadoperez**
- Add support for `watchLiquidations`, `watchLiquidationsForSymbols`, `watchMyLiquidations`, `watchMyLiquidationsForSymbols`.
- Add tests for `fetchLiquidations` and `fetchMyLiquidations` and watch functions

Note:
- I was not able to test the private liquidations, as I couldn't find enough volatility on the testnet.
- I commented the watch tests as I didn't see updates continuos enough and thought it would fail a lot in ci/cd

### CODE REVIEW

File: ts/src/base/Exchange.ts [hunk 1]
```diff
@@ -1601,6 +1611,36 @@ export default class Exchange {
         throw new NotSupported (this.id + ' fetchTradesWs() is not supported yet');
     }
 
+    async watchLiquidations (symbol: string, since: Int = undefined, limit: Int = undefined, params = {}) {
+        throw new NotSupported (this.id + ' watchLiquidations() is not supported yet');
+    }
+
+    async watchAllLiquidations (symbols: string[], since: Int = undefined, limit: Int = undefined, params = {}) {
```
Comment (carlosmiei) [inline ts/src/base/Exchange.ts:1618]: @pcriadoperez should we rename it to `watchLiquidationsForSymbols` to make it compatible with the other implementations?

Comment (carlosmiei) [reply ts/src/base/Exchange.ts:1618]: nevermind just saw you already have it

File: ts/src/base/Exchange.ts [hunk 2]
```diff
@@ -1601,6 +1611,36 @@ export default class Exchange {
...
+        throw new NotSupported (this.id + ' watchAllLiquidations() is not supported yet');
+    }
+
+    async watchLiquidationsForSymbols (symbols: string[], since: Int = undefined, limit: Int = undefined, params = {}) {
```
Comment (carlosmiei) [inline ts/src/base/Exchange.ts:1622]: @pcriadoperez can we add proper return types using the liquidation interface? (we should try to add this type in python as well)

File: ts/src/pro/binance.ts [hunk 3]
```diff
@@ -133,6 +141,375 @@ export default class binance extends binanceRest {
...
+         * @param {int} [limit] the maximum number of liquidation structures to retrieve
+         * @param {object} [params] exchange specific parameters for the bitmex api endpoint
+         * @param {int} [params.until] timestamp in ms of the latest liquidation
+         * @param {boolean} [params.paginate] default false, when true will automatically paginate by calling this endpoint multiple times. See in the docs all the [availble parameters](https://github.com/ccxt/ccxt/wiki/Manual#pagination-params)
+         * @returns {object} an array of [liquidation structures]{@link https://github.com/ccxt/ccxt/wiki/Manual#liquidation-structure}
+         */
+        return this.watchAllLiquidations ([ symbol ], since, limit, params);
+    }
+
+    async watchAllMyLiquidations (symbols: string[], since: Int = undefined, limit: Int = undefined, params = {}) {
```
Comment (carlosmiei) [inline ts/src/pro/binance.ts:408]: I'm not 100% convinced yet if `watchAll` is the best prefix

Comment (pcriadoperez) [reply ts/src/pro/binance.ts:408]: I agree, I also don't like having two methods that are named different but do the same. Maybe I should delete watchAll and keep for symbols?

**Comment by pcriadoperez** [comment]
- Tested binance watchLiquidations ts/py/php/cs ✅ 
- Tested okx watchLiquidations ts/py/php/cs ✅ 
- Bybit was not able to get response ❌ 
- Bitmex was not able to get a response ❌ 
- watchMyLiquidations was not able to trigger ❌

**Pull request merged**